content/docs/interactive/javascript.md

```diff
@@ -0,0 +1,197 @@
+
+## Range Query
+
+Let's now query the dataset to get all rooms between prices 50 to 100.
+We'll use appbase-js [`search()`](https://docs.appbase.io/javascript/api-reference.html#search) method to fire a [range query](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-range-query.html) passing the field `price` and mentioning [range filter](https://www.elastic.co/guide/en/elasticsearch/reference/1.6/query-dsl-range-filter.html) values.
+
+<br/>
+
+<iframe height="600px" width="100%" src="https://repl.it/@dhruvdutt/Appbaseio-Nodejs-range?lite=true" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>
+
+[Here's](https://opensource.appbase.io/mirage/#?input_state=XQAAAAJfCAAAAAAAAAA9iIhnNAWbsswtYjeQNZkpzQK4_mOzUeDpWmI_q1W5J_v7Zsy4Ujaaw71A1BS9rYYbaidH1ngBtQ-I1sDSXRgrmGsCDzBYBoUXwDHQtefpH-PChYyyKqpnVdVrmIsxvIDhOBtThtu_W53GnLmSMoma1UPnh9E7LZRkgXxp3ltXA31wX1fcfowk1r2gVrCN8VgmuPFOWM3o65_HcKkYs4OQ0hAB7hnHy3CILQ5MgAbYZpuCAVHzQcRXBvN2fFZCuCSUNYX32cshZ3d4lrzfi6fQXgJyuLDjbofRzIiFdNbnDTxOh7awm5G8KyWubtxOjtXV8bilwzwGRcwaa6Vwl0TchA1_BpzJh_fwAq12FFL4R-I5_ibh6CZnEvt4-KnlIuy77b7P8zLtJY5p9O0p5F9in_I1bpUcoSjIVyDVUw7CO3XhqV8GV08WBceOu8bcrsKmeZnI-Tz7wJnFBanEKKCvxmdgNX5iWOlyqb-KulI-V5b7Kd0MnlaLDE9KV4rJlTuHN38cafRSw9KFDmA369s6KHGfK4E8tjcHhkopCEWlojhNy-YDYrfkpB4hBkSRnaoc-Ehh6YMq5B_KZpn0W096kfKkWqUYOqjJkBPWLydygFC5kxq4739mGAfM8pTAFnMN5Qt7qFqbuST9fmTei6v0lp3RT2Uxd9LVAqm3Zwpe8RBtPFYbcd_j8RuCqd4O1HeEgZz60qAFmYL37FBNNLpbxT4mhagGqPSzB6jDCruEgDwj6598PSwxTs3SBIqdZtuv_ZTluhF7jvFMerbm7dw7E8yu-B3jNfPoGW9bTfiMthMn620XFd8bmenoEP8pWq4A) the same query executed on Mirage.
+
+## Search Stream
+
+Let's now subscribe to some streaming data based on a query For this, we'll use [`searchStream()`](http://localhost:8000/javascript/api-reference.html#searchstream) method which continuously stream new results of the subscribed query on a give `type`. Here, we're querying with `match_all` to get streaming updates about all data.
+
+We'll also need to make a change on the `type` we've subscribed. For adding some data, we'll use [`index()`](http://localhost:8000/javascript/api-reference.html#index) method and pass a new data object.
+
+<br/>
+
+<iframe height="600px" width="100%" src="https://repl.it/@dhruvdutt/Appbaseio-Nodejs-searchStream-index?lite=true" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>
+
+<!-- Mirage: stream query note? -->
+
+
+## GeoDistance Query
+
+TODO: Desc
+
+Get all the properties around a particular (lat,lon) co-ordinate.
```

Comment:
Tried [this](https://opensource.appbase.io/mirage/#?input_state=XQAAAAKmCAAAAAAAAAA9iIhnNAWbsswtYjeQNZkpzQK4_mOzUeDpWmI_q1W5J_v7Zsy4Ujaaw71A1BS9rYYbaidH1ngBtQ-I1sDSXRgrmGsCDzBYBoUXwDHQtefpH-PChYyyKqpnVdVrmIsxvIDhOBtThtu_W53GnLmSMoma1UPnh9E7LZRkgXxp3ltXA31wX1fcfowk1r2gVrCN8VgmuPFOWM3o65_HcKkYs4OQ0hAB7hnHy3CILQ5MgAbYZpuCAVHzQcRXBvN2fFZCuCSUNYX32cshZ3d4lrzfi6fQXgJyuLD2vgQns3dUESCtKWJx7m6L3fxjViX04mcxbte-NrsTOVNx3X2YvOd2UG_O3P5pgvLecXNPgnsZACPDmsU8OKslnavslVZEjTn_TFj6xc7bgYqdvYkOK1-UKMgOKP0J3jYoVLUimxuXXxXt6dP_ra5LFnfLugV6q2FimZ68iEFmUQu3DhAd_k8ND5N0bJvuBck1dR1iEAz6RDOVmkEhoPiChWYu13ode1Gg0LaGwxR7f5Sw6bXH7iQVo-tF9ZNLgxWzDusXVR1LhIE5FDRzAi-sirhi_O1gVKDhgz56fNmhVGQA26KqVbqsJxF9kCYb8gLRUCMdX5Taj4cuL7Z9JlWm7slNgYLW82a5MXksCXeFSNJ_p5wqkkf5pUCEx4S5xJ-ZW_ukRSv6opCd5QrbnYAer2xySyVSwlqEWicJYXluRI0aHNWfOyZxdEI_XqCkXLAJ9_wIQj1PC-vLwaaHJ2CAQ3algt-iMjUc4E6oR3IxBxH-VZ8p9_OCTVR311kTAwoKVKUyNpGDoc_uhjOTIdM5DDB8uQI5VkQ5Zrt-y81p8c5qM_90vYYA) on Mirage but not entirely sure what we need here.

Can you please link me an example query?

Reply:
Something like [this](https://opensource.appbase.io/mirage/#?input_state=XQAAAAK0CAAAAAAAAAA9iIhnNAWbsswtYjeQNZkpzQK4_mOzUeDpWmI_q1W5J_v7Zsy4Ujaaw71A1BS9rYYbaidH1ngBtQ-I1sDSXRgrmGsCDzBYBoUXwDHQtefpH-PChYyyKqpnVdVrmIsxvIDhOBtThtu_W53GnLmSMoma1UPnh9E7LZRkgXxp3ltXA31wX1fcfowk1r2gVrCN8VgmuPFOWM3o65_HcKkYs4OQ0hAB7hnHy3CILQ5MgAbYZpuCAVHzQcRXBvN2fFZCuCSUNYX32cshZ3d4lrzfi6fQXgJyuLD2vgQns3dUESCtKWJx7m6L3fxjViX04mcxbte-NrsTOVNx3X2YvOd2UG_O3P5pgvLecXNPgnsZACPDmsU8OKslnavslVZEjTn_TFj6xc7bgYqdvYkOK1-UKMgOKP0J3jYoVLUimxuXXxXxgHLnuSBlYv4MbQi1Hw9WjzpILKsFOhJNVGk3h88itDk2RILhqrJIU74eps4pG2P_9U-HhoWZ4vJXrsMx4sgFLKwuxL9JnpcoI8I9h3KCvxj2JVBPKrS2EACCLhtc-_1rz1FVJZZFtuHP4mvbmoYOWEpxOb6adSXi4Pf2DMuwPSBUTDVUc0JFmAgbj3uc87aSFzvbyhabjxPcjthDQCIsXfXwVAEr_-T-YQ1Q_VMvfGlPPq-k6ze89d9Y593K_HoB7G2WX35A-ha3oTTld7jcHFuKjX_042jNudf8e2uIjQgbhAD9icDO55t5cbJk3t0zBzAAI7W9Yk8ggDY889WnR4tLL1aO669LO3QpZGOZzzLVTDTSurgTi9FKtwI5397AOTlz00ZTc_FeeH4l5syXeqGQnXoBRvcoFWau0mfdJfQin__2WFBm) and a distance added as `5mi`.